scoap3/modules/tools/tools.py

```diff
@@ -39,74 +37,69 @@ def affiliations_export(country=None, year=None):
     result_headers = ['year', 'journal', 'doi', 'arxiv number', 'primary arxiv category',
                       'country', 'affiliation', 'authors with affiliation', 'total number of authors']
     result_data = []
-    index = 0
 
     # query ElasticSearch for result (and get total hits)
     query = get_query_string(country=country, year=year)
     search_results = current_search_client.search(
-        q=query, index=search_index, _source=source_fields, size=size, from_=index
-    )
+        q=query, index=search_index, _source=source_fields, scroll='1m')
+
+    sid = search_results['_scroll_id']
+    scroll_size = len(search_results['hits']['hits'])
 
-    total_hits = search_results['hits']['total']['value']
     logger.info('Searching for affiliations of country: {} and year: {}'.format(
         country if country else 'ALL',
         year if year else 'ALL'
     ))
     logger.info('Total results from query: {}'.format(total_hits))
 
-    if total_hits == 0:
+    if scroll_size == 0:
         return {'header': result_headers, 'data': result_data}
 
-    while index < total_hits:
+    while (scroll_size > 0):
+        for record_index in range(scroll_size):
```

Comment:
you just need  iterate through the results:

```
for result in search_results['hits']['hits']:
 ....
 
 ```
 
 
 this way you don't need to keep an index